Makefile

```diff
@@ -51,13 +53,29 @@ test-coverage:
 	poetry run coverage report --show-missing
 	poetry run coverage xml
 
+generate-man:
+	@export QPC_VAR_CURRENT_YEAR=$(shell date +'%Y') \
+	&& export QPC_VAR_PROJECT=$${QPC_VAR_PROJECT:-Quipucords} \
+	&& export QPC_VAR_PKG_SOURCE=$${QPC_VAR_PKG_SOURCE:-qpc} \
+	&& jinja -X QPC_VAR docs/source/man.j2 $(ARGS)
+
+update-man.rst:
+	$(MAKE) generate-man ARGS="-o docs/source/man.rst"
+
+manpage-test:
+	@poetry run $(MAKE) generate-man | grep -v "Entering directory" | \
```

Comment:
I have an idea that may negate the need for these negative greps. Putting this comment here as a placeholder until I have tested my idea so we don't forget.

Reply:
Actually, now that I try it locally, I don't think _any_ of the `grep`s are necessary. 🤔

I changed this locally:

```diff
diff --git a/Makefile b/Makefile
index b5ca073..6ee028b 100644
--- a/Makefile
+++ b/Makefile
@@ -63,13 +63,11 @@ update-man.rst:
        $(MAKE) generate-man ARGS="-o docs/source/man.rst"

 manpage-test:
-       @poetry run $(MAKE) generate-man | grep -v "Entering directory" | \
-       grep -v "Leaving directory" > output.txt
-       @diff -u docs/source/man.rst output.txt
+       @poetry run $(MAKE) generate-man | diff -u docs/source/man.rst -

 manpage:
-       @$(MAKE) generate-man | grep -v "Entering directory" | \
-       grep -v "Leaving directory" | \
+       @$(MAKE) generate-man | \
        $(pandoc) -s - \
          --standalone -t man -o docs/$(QPC_VAR_PKG_SOURCE).1\
          --variable=section:1\
```

But I do not see "Entering directory" or "Leaving directory" anywhere in the output or generated files. Maybe it used to appear but no longer appears due to some other change? I recommend removing the `grep`s like this and double-check that the commands still complete as expected.

(Note that I also removed the intermediary `output.txt` file so that running these commands will not leave a dirty/untracked file in the local repository.)

Originally, when I assumed that "Entering directory" or "Leaving directory" was still being generated somewhere, I was going to suggest changing how we call `make` from within `make`, because `make` _itself_ would be responsible for those messages. The `--no-print-directory` argument would tell `make` to silence these specific informational messages. Alternatively, the `-s` argument would tell `make` to silence _all_ make-related messages.

I wonder if you and I have something different in our environments that is causing `make` to emit the directory changes for you but not for me. I wonder if you have any `MAKEFLAGS` set in your shell environment.

Reply:
Mmm worth a shot. I did reproduce a issue she was having locally and suggested writing to a temporary file to avoid these negative greps. I prefer your solution, though 👍🏽

Reply:
I will push the changes.

Reply:
ok, I will try to make it work locally based on your suggestion.

Reply:
Just confirming that "Entering directory" or "Leaving directory" messages are still being generated for me locally, I am not sure why, I have no `MAKEFLAGS` set on my shell. Since this behavior was also reproduced locally in @bruno-fs 's env I believe the best approach would be calling the `make `command with the `--no-print-directory`, that works well. You mentioned that -s would be an alternative, but wouldn't this flag also suppress any error messages related to the command generate-man ? In the end we would still reach our goal and have the test fail, but wouldn't it be interesting to see why the command failed in the first place ?

Reply:
You are correct. `--no-print-directory` is probably the better choice here because, yes, `-s` may more aggressively suppress other messages we do want to see.